source/extensions/filters/http/grpc_stats/message_counter.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "envoy/buffer/buffer.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace GrpcStats {
+
+// The struct to store gRPC message counter state.
+struct GrpcMessageCounter {
+  // gRPC uses 5 byte header to encode subsequent message length
+  enum GrpcReadState {
```

Comment:
Per discussion let's move this next to `Grpc::Frame` and try to figure out how to share some logic.

Reply:
Done. Merged this utility into the main codec.